cpp/harvest/nacjd.cc

```diff
@@ -158,26 +165,52 @@ void ExtractExistingIDsFromMarc(MARC::Reader * const marc_reader, std::set<std::
     }
 }
 
+void ExtractOneIDFromWebsite(const unsigned nacjd_id) {
+    std::string nacjd_title_path("/tmp/nacjd_titles.html");
+    std::string nacjd_JSON_path("/tmp/nacjd_new_titles.json");
+
+    if (FileUtil::Exists(nacjd_title_path))
+        FileUtil::DeleteFile(nacjd_title_path);
+
+    FileUtil::WriteString(nacjd_title_path, "");
+
+    std::ifstream file(nacjd_title_path);
+    if (file)
+        std::for_each(std::istream_iterator<char>(file), std::istream_iterator<char>(), HandleChar);
+    else
+        LOG_ERROR("couldn't open file: " + nacjd_title_path);
+    if (FileUtil::Exists(nacjd_JSON_path) and not FileUtil::DeleteFile(nacjd_JSON_path))
+        LOG_ERROR("Could not delete file: " + nacjd_JSON_path);
+
+    std::ofstream json_new_titles(nacjd_JSON_path);
+
+    json_new_titles << "{ \"nacjd\" : [ " << '\n';
+    bool first(true);
+
+    DownloadID(json_new_titles, std::to_string(nacjd_id), /*use_separator*/ not first);
+
+    json_new_titles << " ] }";
+}
 
 void ExtractIDsFromWebsite(const std::set<std::string> &parsed_marc_ids, unsigned * const number_of_new_ids, const unsigned count_rows) {
     const std::string DOWNLOAD_URL(
         "https://www.icpsr.umich.edu/web/NACJD/search/"
         "studies?start=0&ARCHIVE=NACJD&PUBLISH_STATUS=PUBLISHED&sort=DATEUPDATED%20desc&rows="
         + std::to_string(count_rows));
-    if (FileUtil::Exists(NACJD_TITLES))
-        FileUtil::DeleteFile(NACJD_TITLES);
+    if (FileUtil::Exists("/tmp/nacjd_titles.html"))
+        FileUtil::DeleteFile("/tmp/nacjd_titles.html");
 
-    if (not Download(DOWNLOAD_URL, NACJD_TITLES, TIMEOUT_IN_SECONDS * 1000))
+    if (not Download(DOWNLOAD_URL, "/tmp/nacjd_titles.html", TIMEOUT_IN_SECONDS * 1000))
         LOG_ERROR("Could not download website with nacjd ids.");
-    std::ifstream file(NACJD_TITLES);
+    std::ifstream file("/tmp/nacjd_titles.html");
     if (file)
         std::for_each(std::istream_iterator<char>(file), std::istream_iterator<char>(), HandleChar);
     else
-        LOG_ERROR("couldn't open file: " + NACJD_TITLES);
+        LOG_ERROR("couldn't open file: /tmp/nacjd_titles.html");
 
-    if (FileUtil::Exists(NACJD_NEW_TITLES_JSON) and not FileUtil::DeleteFile(NACJD_NEW_TITLES_JSON))
-        LOG_ERROR("Could not delete file: " + NACJD_NEW_TITLES_JSON);
-    std::ofstream json_new_titles(NACJD_NEW_TITLES_JSON);
+    if (FileUtil::Exists("/tmp/nacjd_new_titles.json") and not FileUtil::DeleteFile("/tmp/nacjd_new_titles.json"))
```

Comment:
The path to the JSON output file should in this case be passed as a function parameter to `ExtractIDsFromWebsite`. When it is called in the `getAll` function, we can also generate an AutoTempFile there for now which will then be passed to `ParseJSONAndWriteMARC`. 

Of course, in the long term we want to remove this step entirely from this mode and add a `convert` mode instead, so the other option would be to just use argv[3] as JSON target path.